File: ruby/lib/steam/community/tf2/tf2_medic.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2008-2011, Sebastian Staudt

require 'steam/community/tf2/tf2_class'

# Represents the stats for the Team Fortress 2 Medic class for a specific user
#
# @author Sebastian Staudt
class TF2Medic

  include TF2Class

  # Returns the maximum health healed for teammates by the player in a
  # single life as a Medic
  #
  # @return [Fixnum] Maximum health healed
  attr_reader :max_health_healed

  # Returns the maximum number of ÜberCharges provided by the player in a
  # single life as a Medic
  #
  # @return [Fixnum] Maximum number of ÜberCharges
  attr_reader :max_ubercharges

  # Creates a new instance of the Medic class based on the given XML data
  #
  # @param [REXML::Element] class_data The XML data for this Medic
  def initialize(class_data)
    super class_data

    @max_health_healed = class_data.elements['ihealthpointshealed'].text.to_i
    @max_ubercharges   = class_data.elements['inuminvulnerable'].text.to_i
  end

end

File: php/lib/steam/community/tf2/TF2Medic.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2008-2009, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Steam Community
 */

require_once "steam/community/tf2/TF2Class.php";

/**
 * Represents the stats for the Team Fortress 2 medic class for a specific user
 * @package    Steam Condenser (PHP)
 * @subpackage Steam Community
 */
class TF2Medic extends TF2Class
{
	/**
	 * Creates a new instance of TF2Medic based on the assigned XML data
	 * @param $classData
	 */
	public function __construct($classData)
	{
		parent::__construct($classData);

		$this->maxUberCharges    = (int) $classData->inuminvulnerable;
		$this->maxHealthHealed   = (int) $classData->ihealthpointshealed;
	}
}
?>
